pkg/coordinator/coordinator.go

```diff
@@ -210,23 +232,24 @@ func (rc *reportCoordinator) checkLogs() {
 			continue
 		}
 
-		logCheckBlockKey, id, err := l.Key.BlockKeyAndUpkeepID()
+		logCheckBlockKey, id, err := rc.enc.SplitUpkeepKey(l.Key)
 		if err != nil {
 			continue
 		}
-		nextKey, err := logCheckBlockKey.Next()
+		nextKey, err := rc.enc.Increment(logCheckBlockKey)
```

Comment:
I'm wondering if we're going slightly beyond the realm of a refactor with this change? But I guess if the integration tests pass it should be grand?

Reply:
The way we're handling some of these values directly make the encoder much more complicated. Maybe we can find a simpler way?